pkg/kn/commands/wait_flags.go

```diff
@@ -42,8 +42,15 @@ type WaitFlags struct {
 func (p *WaitFlags) AddConditionWaitFlags(command *cobra.Command, waitTimeoutDefault int, action, what, until string) {
 	waitUsage := fmt.Sprintf("%s %s and don't wait for it to be %s.", action, what, until)
 	//TODO: deprecated flag should be removed in next release
+
+	// Special-case 'delete' command so it comes back to the user immediately
+	noWaitDefault := false
+	if action == "Delete" {
```

Comment:
I'd rather see default value provided through function's argument. The current approach will change default for all delete operations, e.g. also Revisions. What about a new config struct that for now can have 2 fields, default value and timeout.

Reply:
oh I didn't realize you could be more specific... I'll fix....

Reply:
btw, I would be happy to switch to no-wait as default for all delete operations (this would then be also easy to reason about).

Reply:
I do like consistency :-)  We just need to figure out which way it spans.  Making all deletes the same (as long as it doesn't have negative side-effects) seems ok to me.  @dsimansk ?

Reply:
I wasn't sure about all the usage. but wait flags are used in Service (delete) and Revision (delete), no objections to go with `--no-wait` for all delete then.

Reply:
ok - if we do that, then I think the PR is ready for review